### PR TITLE
Replace Win32 file IO with UWP safe variants and add support for getting drives to UWP build

### PR DESCRIPTION
**Pull request by tunip3**
This PR replaces all Win32 file IO with UWP safe variants (at least for all file io functions in https://docs.microsoft.com/en-us/windows/win32/api/fileapifromapp/).

This PR also implements drive support into the UWP build, this means that it now actually shows all your drives when you go to the root path.

### CODE REVIEW

**Comment by hrydgard** [comment]
Good stuff!

The one other major I/O problem we have on UWP is accessing the ISO on the main thread, but maybe it's no longer an issue. That's what created the need for the horrible UWP/StorageFileLoader.cpp. Feel free to see if that can be gotten rid of too in a followup if you like, otherwise I'll take a look relatively soon.

One blocker: You accidentally added the giant file UWP/NATIVEAPP.ipch, we should probably make sure gitignore takes care of it - please remove it and force-push it away so it doesn't get into the PPSSPP repo.

File: Common/File/DirListing.cpp [hunk 1]
```diff
@@ -308,7 +329,7 @@ std::vector<std::string> GetWindowsDrives()
 
 			// advance to next drive
 			while (*drive++) {}
-		}
+}
```
Comment (unknownbrackets) [inline Common/File/DirListing.cpp:332]: This looks unintentional.

-[Unknown]

Comment (tunip3) [reply Common/File/DirListing.cpp]: fixed

**Comment by hrydgard** [comment]
Thanks. Will test this tonight, then squash-merge it (so the big file won't end up in history).

**Comment by hrydgard** [comment]
Tested this on both Windows and Xbox, worked just fine!

Gonna have to make a faster way to get to the drives root from the default dir, but very nice that it's possible. Thanks!

**Review by hrydgard — APPROVED** [review]

**Pull request merged**